src/uwtools/resources/rocoto.jsonschema

```diff
@@ -161,7 +161,18 @@
           "type": "object"
         },
         "^timedep(_.*)?$": {
-          "$ref": "#/$defs/TimePattern"
+          "additionalProperties": false,
+          "properties": {
+            "attrs": {
+              "additionalProperties": false,
+              "properties": {
+                "offset": {
+                  "type": "string"
+                }
+              },
```

Comment:
I updated the schema to expand on the `timedep` tag. Here we need to take into account for the `offset` key within `timedep`. I also added `attrs` to `timedep` which is what we talked about UW Group this week.

I did not update `datadep` as it already had the `attrs` item within the schema and had all the appropriate tags needed.

Reply:
"...`timedep` is just an alias (ref) for compoundTimeString. Do we have a representation for that in JSON Schema yet?"

Yes we have `timedep` referencing the `TimePattern` definition in the JSONSchema where it matches a specific regex expression for `^(\\d{2}:){3}\\d{2}$`. I don't believe this covers any arbitrary text string, and in the JSONSchema we can only reference one definition in this case`"$ref": "#/$defs/TimePattern"`.

Reply:
I don't think we have to deal with it in this PR, but here's what Rocoto actually allows for `compoundTimeString`: https://github.com/christopherwharrop/rocoto/blob/develop/lib/workflowmgr/schema_with_metatasks.rng#L203-L210 -- i.e. any combination of one or more blocks of text and `<cyclestr>` elements. The `<cyclestr>` element itself has an `offset` attribute, which has a value that may be what our `TimePattern` is trying to emulate. Rocoto's spec for the `time` pattern also looks somewhat more complicated/flexible (https://github.com/christopherwharrop/rocoto/blob/develop/lib/workflowmgr/schema_with_metatasks.rng#L181-L188) than what we have for `TimePattern`.

It's not clear to me what the pieces of text inside `compoundTimeString` are supposed to be. As far as I can tell from the RELAX NG schema, they could be anything at all, though that seems weird. I mean, would Rocoto really accept, say,
```
<dependency>
  <timedep>
    Why
    <cyclestr offset="12:05">Hello</cyclestr>
    There
  </timedep>
</dependency>
```
It looks like its schema would accept that, but maybe Rocoto would crash. Who knows? Seems weird. The Rocoto docs say
> The value between the start and end of the <timedep> tag is a time in yyyymmddhhmmss format.
but I don't think that's enforced by its schema, if I'm reading the RELAX NG code correctly. And the example code where that quote appears is
```
<timedep><cyclestr offset="&DEADLINE;">@Y@m@d@H@M@S</cyclestr></timedep>
```
so I'm not sure `<timedep>` is really meant to be valid without `<cyclestr>`, whatever the schema says.

I think for this PR a `<timedep>` that accepts whatever text content should be fine. But it doesn't have an `offset` attribute as far as I can tell -- that belongs to the `<cyclestr>` tag, and we'll have to deal with that soon enough, as there's a separate Jira ticket for that. But our JSON Schema `TimePattern` is definitely not the same thing as the RELAX NG schema's `compoundTimeString`.

Reply:
Yes I agree, the `TimePattern` is distinctly different than the `compoundTimeString`. Currently we have ticket UW-386, which I think would need to address much of the needs you just outlined for us.

Then I think what we are talking is if we want `timedep` to accept any text content than we would need to change the schema to use a reference definition of `anyStringPattern`. For example I would think we would update the lines highlight on this comment to be:

```
          "additionalProperties": false,
          "properties": {
            "attrs": {
              "additionalProperties": false,
                  "$ref": "#/$defs/anyStringPattern"
              },
```

where `anyStringPattern` would be:
```
    "anyStringPattern": {
      "pattern": ".+", --> This would match any string content as long as it has 1 character in it
      "type": "string"
    },
```

Reply:
Looking at the json schema there are only three pre-defined definitions that are `TimePattern`, `datePattern`, `dateTimePatten`. 

Another approach here would be leaving it as:
```
        "^timedep(_.*)?$": {
          "type": "string"
        }
```
This would mean that whatever `attrs` are within `timedep` will just be treated as a single string. Seemed to be the most simple way to handle this case.

Reply:
Also if we decide to go this route, the `TimePattern` definition is only being used by the `timedep` tag. It might be worth leaving changing `timedep` to the above example, and consider remving `TimePattern` for now, which could in the future come back pending how `<cyclestr>` is handled.

Reply:
I like this idea, because I think `TimePattern` may already not be fully compatible with the RELAX NG schema's definition. If `TimePattern` isn't used elsewhere, I like the simplification of removing it now and then re-implementing it when we add `<cyclestr>` support, which will need it. We can then revisit the RNG schema to guide implementation. At that point, I think we should also consider just making our JSON Schema `timedep` a reference to whatever we call our implementation of `compoundTimeString`, which is how the RNG schema defines it.

Reply:
I like this idea too. It gives more flexibility on how to handle `<cyclestr>` later to who will be working the ticket covering this discussion.